src/prefect/engine/task_runner.py

```diff
@@ -893,28 +889,27 @@ def get_task_run_state(
         except signals.LOOP as exc:
             new_state = exc.state
             assert isinstance(new_state, Looped)
-            new_state.result = Result(
-                value=new_state.result, result_handler=self.result_handler
+            new_state.result = self.task.result.write(
+                new_state.result, **prefect.context
             )
             new_state.cached_inputs = inputs
             new_state.message = exc.state.message or "Task is looping ({})".format(
                 new_state.loop_count
             )
             return new_state
 
-        result = Result(value=result, result_handler=self.result_handler)
+        result = self.task.result.write(return_value, **prefect.context)
         state = Success(
             result=result, message="Task run succeeded.", cached_inputs=inputs
         )
 
-        ## checkpoint tasks if a result_handler is present, except for when the user has opted out by disabling checkpointing
+        # checkpoint tasks if a result_handler is present, except for when the user has opted out by disabling checkpointing
         if (
             state.is_successful()
             and prefect.context.get("checkpointing") is True
             and self.task.checkpoint is not False
-            and self.result_handler is not None
         ):
-            state._result.store_safe_value()
+            state._result.store_safe_value()  # todo: here
```

Comment:
In the discussion PR @cicdw mentions maybe this whole idea goes away since we are greedily writing results already and the serializers are also already going to take care of sending the filepath (https://github.com/PrefectHQ/prefect/pull/2278#discussion_r404316357); at one point we had an IRL discussion about maybe checking in or during `.write()` whether checkpointing is off or not if we still want to turn this off. Thoughts? How does it play with cloud's checkpoint-all-the-time feature?